### ISSUE TITLE
cannot install scikit-bio on fresh pyenv environment

### ISSUE THREAD

**Issue by jsevo** [biocore/scikit-bio#1600]
I am using pyenv, python version Python 3.7.0. Creating a new environment and installing numpy first yields:

Command "/Users/user/.ve/skbio/bin/python3.7 -m pip install --ignore-installed --no-user --prefix /private/var/folders/4y/79v_6xrd6yj4v8ddkrf2zbc40000gn/T/pip-build-env-f5fdlm8s --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- wheel setuptools Cython "numpy==1.9.3; python_version=='3.5'" "numpy==1.12.1; python_version=='3.6'" "numpy==1.13.1; python_version>='3.7'"" failed with error code 1 in None

**Comment by mortonjt** [comment]
There are some issue with python=3.7 that haven't been addressed yet.  In the meantime, I'd recommend using python=3.5.  This can be done via conda installation i.e.

`conda create -n skbio python=3.5 scikit-bio -c conda-forge`

**Comment by mortonjt** [comment]
more discussion this can be found here: https://github.com/biocore/scikit-bio/issues/1595

**Comment by mortonjt** [comment]
We are having a sprint on this soon and will be discussing this.  Stay tuned!

**Comment by Benjamin-Lee** [comment]
Is there any news on this? If nothing else, can we at least get a note on the README that explains that the software doesn't support 3.7?

**Comment by thermokarst** [comment]
Good news! The 0.5.5 release supports python 3.7 --- https://github.com/biocore/scikit-bio/blame/master/CHANGELOG.md#L44.

**Comment by thermokarst** [comment]
@Benjamin-Lee - are you experiencing an issue running scikit-bio in a py3.7 environment?

**Comment by Benjamin-Lee** [comment]
Yes but I think I fixed it (not entirely sure why my fix worked though). I was installing from a requirements.txt and getting a numpy not installed error but when I manually pip installed numpy and then scikit-bio it worked. No idea what the issue is but it's working now.

Thanks for the prompt response!

**Comment by jdiggans-twist** [comment]
We are having this same issue - it appears to be related to the import of numpy in scikit-bio's `setup.py`, see: https://github.com/biocore/scikit-bio/blob/master/setup.py#L20

This means that any (custom) package depending on scikit-bio must pre-install numpy (outside of the `python setup.py test` command); not doing this results in `python setup.py test` (for the custom package) failing for lack of numpy when it tries to install scikit-bio.

Import of third-party packages in setup.py, especially complicated packages like numpy, is not something I'd describe as a best practice. Does this have to be in setup.py for scikit-bio?

**Comment by ElDeveloper** [comment]
@jdiggans-twist, as far as I can tell the import is only there to find the path to the NumPy header files. Seems like this might be a way around it:
https://stackoverflow.com/a/54153319/379593

**Comment by jdiggans-twist** [comment]
Thanks @ElDeveloper - I figured a way around it; per https://github.com/numpy/numpy/issues/13430 this is caused by an issue between Cython and recent versions of numpy. Backing off the numpy version (to 1.13.x in my case, as it was the last known good version) solved this.

**Comment by oktaykaplan** [comment]
I am having the same problem and cannot install scikit-bio on Mac (Python 3.7). I have tried a number of suggestions but failed to install it.  Any suggestion?

**Comment by ElDeveloper** [comment]
What's the error you are seeing? If you aren't already installing using conda is likely the easiest way, I just tried this out in a fresh environment and it worked well.

**Comment by oktaykaplan** [comment]
Dear Yoshiki,

Thank you for your kind help. Let me share the code I have tried with you.
(Python 3.7.0 Jupyter Notebook and Mac).

Best regards
Oktay


Failed to build scikit-bio hdmedians
Installing collected packages: hdmedians, scikit-bio
    Running setup.py install for hdmedians ... error    ERROR: Command
errored out with exit status 1:
     command: /Users/OKTAYMAC/anaconda3/bin/python -u -c 'import sys,
setuptools, tokenize; sys.argv[0] =
'"'"'/private/var/folders/y4/5tjrtmh13z3dglyhnc_hxt400000gn/T/pip-install-kvn3afw_/hdmedians/setup.py'"'"';
__file__='"'"'/private/var/folders/y4/5tjrtmh13z3dglyhnc_hxt400000gn/T/pip-install-kvn3afw_/hdmedians/setup.py'"'"';f=getattr(tokenize,
'"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"',
'"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))'
install --record
/private/var/folders/y4/5tjrtmh13z3dglyhnc_hxt400000gn/T/pip-record-_cyxm6il/install-record.txt
--single-version-externally-managed --compile --install-headers
/Users/OKTAYMAC/anaconda3/include/python3.7m/hdmedians
         cwd: /private/var/folders/y4/5tjrtmh13z3dglyhnc_hxt400000gn/T/pip-install-kvn3afw_/hdmedians/
    Complete output (20 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.7-x86_64-3.7
    creating build/lib.macosx-10.7-x86_64-3.7/hdmedians
    copying hdmedians/__init__.py -> build/lib.macosx-10.7-x86_64-3.7/hdmedians
    copying hdmedians/medoid.py -> build/lib.macosx-10.7-x86_64-3.7/hdmedians
    running build_ext
    skipping 'hdmedians/geomedian.c' Cython extension (up-to-date)
    building 'hdmedians.geomedian' extension
    creating build/temp.macosx-10.7-x86_64-3.7
    creating build/temp.macosx-10.7-x86_64-3.7/hdmedians
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG
-g -fwrapv -O3 -Wall -Wstrict-prototypes
-I/Users/OKTAYMAC/anaconda3/include -arch x86_64
-I/Users/OKTAYMAC/anaconda3/include -arch x86_64
-I/Users/OKTAYMAC/anaconda3/lib/python3.7/site-packages/numpy/core/include
-I/Users/OKTAYMAC/anaconda3/include/python3.7m -c
hdmedians/geomedian.c -o
build/temp.macosx-10.7-x86_64-3.7/hdmedians/geomedian.o
    dyld: Library not loaded: @rpath/libswiftCore.dylib
      Referenced from:
/Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild
      Reason: image not found
    gcc: error: unable to locate xcodebuild, please make sure the path
to the Xcode folder is set correctly!
    gcc: error: You can set the path to the Xcode folder using
/usr/bin/xcode-select -switch
    error: command 'gcc' failed with exit status 71
    ----------------------------------------ERROR: Command errored out
with exit status 1: /Users/OKTAYMAC/anaconda3/bin/python -u -c 'import
sys, setuptools, tokenize; sys.argv[0] =
'"'"'/private/var/folders/y4/5tjrtmh13z3dglyhnc_hxt400000gn/T/pip-install-kvn3afw_/hdmedians/setup.py'"'"';
__file__='"'"'/private/var/folders/y4/5tjrtmh13z3dglyhnc_hxt400000gn/T/pip-install-kvn3afw_/hdmedians/setup.py'"'"';f=getattr(tokenize,
'"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"',
'"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))'
install --record
/private/var/folders/y4/5tjrtmh13z3dglyhnc_hxt400000gn/T/pip-record-_cyxm6il/install-record.txt
--single-version-externally-managed --compile --install-headers
/Users/OKTAYMAC/anaconda3/include/python3.7m/hdmedians Check the logs
for full command output.
Note: you may need to restart the kernel to use updated packages.

In [ ]:

def make_seq():

    length =random.choice(range(45,50))

    return ''.join([random.cho

**Comment by ElDeveloper** [comment]
Looks like you are using conda. I suggest you use: `conda install -c conda-forge  scikit-bio`. Solving building issues using a native python installation can be complicatrd.

**Comment by oktaykaplan** [comment]
Dear Yoshiki,

Thank you for your kind help:) It worked:)
